compositor_common/src/scene.rs

```diff
@@ -0,0 +1,37 @@
+use std::{any::Any, collections::HashMap, rc::Rc};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct VideoId(usize);
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct TransformationRegistryKey(String);
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct Resolution {
+    width: usize,
+    height: usize,
+}
```

Comment:
Isn't `u32` more appropriate here? `Resolution` isn't just for array indexing.  

```suggestion
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
pub struct Resolution {
    width: u32,
    height: u32,
}
```

Reply:
yeah, but resolution is limited by your address space size, not by what fits in 32 bits

Reply:
I'm pretty sure resolution is limited by sth a lot smaller than u32. In the purely semantic sense u32 or different fixed size is more correct because public api and value itself does not represent anything that should change with an architecture, on the other hand when using those values we will almost everywhere need usize, so it will be easier to use.

Reply:
I meant that in our application it's limited by what we can download into ram, which is limited by ptr length. There probably is a much stricter limit on what wgpu can render, but on the CPU side this is going to be used mostly for buffer allocation.